app/controllers/AuditController.scala

```diff
@@ -109,31 +112,48 @@ class AuditController @Inject() (implicit val env: Environment[User, SessionAuth
           case "Assigned" =>
             WebpageActivityTable.save(WebpageActivity(0, anonymousUser.userId.toString, ipAddress, "Visit_Audit", timestamp))
 
-            // HITs are associated with routes at the time of HIT creation
-            // Retrieve the route based on HIT ID
-            val routeId: Option[Int] = AMTRouteAssignmentTable.findRouteByHITId(hitId)
+            // Retrieve the route based on the condition that the worker has been assigned to and the associated unvisited routes
+            var cId = AMTConditionTable.assignAvailableCondition
+
+            if(TurkerTable.getConditionIdByTurkerId(workerId) == None){
+              // No worker was found with this id (implies no condition was assigned) so assign the condition id that has been least used
+              // Select amt_condition.amt_condition_id, count(condition_id) as cnt from amt_assignment Right JOIN amt_condition on (amt_assignment.condition_id = amt_condition.amt_condition_id) group by amt_condition.amt_condition_id order by cnt asc;
```

Comment:
That was the query I used to test if I was getting the correct condition id. I've removed the comment.